### ISSUE TITLE
VariableRebinding: false positive with the pin operator in maps

### ISSUE THREAD

**Issue by jurta** [rrrene/credo#486]
### Environment

* Credo version (`mix credo -v`): 0.8.10
* Erlang/Elixir version (`elixir -v`):
```
Erlang/OTP 20 [erts-9.1] [source] [64-bit] [smp:4:4] [ds:4:4:10] [async-threads:10] [hipe]
Elixir 1.5.2
```
* Operating system: GNU/Linux Ubuntu 16.04

### What were you trying to do?

```elixir
    seq = 1
    %{seq: ^seq} = %{seq: 1}
```

### Expected outcome
No warning, because the pin operator doesn't rebind the variable.

### Actual outcome
```
┃ [F] → Variable "seq" was declared more than once.
```

**Comment by rrrene** [comment]
:+1: Thx 
Thx for reporting! :+1:

**Comment by rrrene** [comment]
Thanks for reporting this 😀  It is now fixed on `master`.

You can try this by setting the Credo dep to

```elixir
{:credo, github: "rrrene/credo"}
```

Please report back if your issue is solved! 👍
